src/lib/routes/admin-api/event.ts

```diff
@@ -188,4 +227,33 @@ export default class EventController extends Controller {
             response,
         );
     }
+
+    async searchEvents(
+        req: IAuthRequest<any, any, any, EventSearchQueryParameters>,
+        res: Response<EventSearchResponseSchema>,
+    ): Promise<void> {
+        const { normalizedLimit, normalizedOffset } = normalizeQueryParams(
+            req.query,
+            {
+                limitDefault: 50,
+                maxLimit: 1000,
+            },
+        );
+
+        const { events, totalEvents } = await this.eventService.searchEvents({
+            ...req.body,
+            offset: normalizedOffset,
+            limit: normalizedLimit,
+        });
+
```

Comment:
This bit here is a little hard to follow. What do you get back from `normalizeQueryParams`? If it's the entire object, couldn't you do something like:

```suggestion
        const normalizedQuery = normalizeQueryParams(
            req.query,
            {
                limitDefault: 50,
                maxLimit: 1000,
            },
        );

        const { events, totalEvents } = await this.eventService.searchEvents(normalizedQuery);

```

But maybe it doesn't work like that? 

Anyway, we should be able to deal with the default value in the schema, but because it's a string, there's no way to set a max value, unfortunately.